types/utils.d.ts

```diff
@@ -78,3 +78,12 @@ export type RecordKeysToLowercase<Input> = Input extends Record<string, unknown>
     ]: Input[Key];
   }
   : Input;
+
+type OmitIndexSignature<T> = {
+  [K in keyof T as string extends K ? never : number extends K ? never : K]: T[K];
+};
+
+/**
+ * HTTP header strings
+ */
+export type HttpHeader = keyof OmitIndexSignature<http.OutgoingHttpHeaders> | (string & {}); // eslint-disable-line @typescript-eslint/ban-types
```

Comment:
Please add a comment to warn for the usage on return value.
This hack is perfectly fine for `input` value but not for `return` value.
https://github.com/microsoft/TypeScript/issues/29729#issuecomment-554927299